internal/core/legacy/services/filesvc/download.go

```diff
@@ -92,8 +106,10 @@ func (svc *FileService) DownloadFile(w io.Writer, file dao.File) (headers map[st
 		headerContentDisposition: headerValueDispositionInline(file.Ext, file.DisplayName),
 		HeaderContentLength:      strconv.FormatInt(file.ByteSize, 10),
 	}
-
-	contentType := mime.TypeByExtension(file.Ext)
+	var buf bytes.Buffer
+	tee := io.TeeReader(reader, &buf)
+	contentType := GetFileContentType(tee)
+	io.ReadAll(tee)
```

Comment:
No need readAll in tee, `buf` just need the first few chars.

Reply:
deleted, adjusted by copy(buf), copy(reader)